### ISSUE TITLE
CNI plugins not initialized, even though they are

### ISSUE THREAD

**Issue by spuder** [hashicorp/nomad#7536]
### Nomad version
Nomad: 0.10.4
Consul: 1.7.0
CNI Plugins: 1.8.4

### Operating system and Environment details
Ubuntu 18.04

### Issue

I have a nomad cluster with CNI plugins 1.8.4 installed as [recommended in the installation guide](https://nomadproject.io/docs/integrations/consul-connect/#cni-plugins). 
I am still seeing error `cni plugin not initialized` when I attempt to stop the example countdash job for testing nomad + consul connect. It is not reproducible every time. 

```
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.799Z [ERROR] client.alloc_runner.runner_hook: failed to cleanup network for allocation, resources may have leaked: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e alloc=6d4b864b-4ad2-6368-e74f-3bba4161148e error="cni plugin not initialized"
```

The plugins are installed and have the correct permissions
```
ll /opt/cni/bin
total 70192
drwxr-xr-x 2 nomad nomad     4096 Jan  9 12:51 ./
drwxr-xr-x 3 nomad nomad     4096 Mar 10 22:39 ../
-rwxr-xr-x 1 root  root   4159253 Jan  9 12:50 bandwidth*
-rwxr-xr-x 1 root  root   4628066 Jan  9 12:50 bridge*
-rwxr-xr-x 1 root  root  12120140 Jan  9 12:51 dhcp*
-rwxr-xr-x 1 root  root   5945760 Jan  9 12:50 firewall*
-rwxr-xr-x 1 root  root   3069556 Jan  9 12:50 flannel*
-rwxr-xr-x 1 root  root   4113837 Jan  9 12:50 host-device*
-rwxr-xr-x 1 root  root   3614305 Jan  9 12:51 host-local*
-rwxr-xr-x 1 root  root   4275320 Jan  9 12:50 ipvlan*
-rwxr-xr-x 1 root  root   3205277 Jan  9 12:50 loopback*
-rwxr-xr-x 1 root  root   4346248 Jan  9 12:50 macvlan*
-rwxr-xr-x 1 root  root   3896326 Jan  9 12:50 portmap*
-rwxr-xr-x 1 root  root   4546820 Jan  9 12:50 ptp*
-rwxr-xr-x 1 root  root   3392736 Jan  9 12:50 sbr*
-rwxr-xr-x 1 root  root   2885430 Jan  9 12:51 static*
-rwxr-xr-x 1 root  root   3356497 Jan  9 12:50 tuning*
-rwxr-xr-x 1 root  root   4275168 Jan  9 12:50 vlan*
```

```
$ echo 1 > /proc/sys/net/bridge/bridge-nf-call-arptables
$ echo 1 > /proc/sys/net/bridge/bridge-nf-call-ip6tables
$ echo 1 > /proc/sys/net/bridge/bridge-nf-call-iptables
```

I have stopped and restarted the countapi 10+ times and the error `cni plugin not initialized`
has only been reproducible once. It seems like a race condition. 

```

Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020/03/28 18:20:19.073980 [INFO] (runner) starting
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020/03/28 18:20:19.077102 [INFO] (runner) rendered "(dynamic)" => "/opt/nomad/alloc/e1ebf603-838a-a1bd-54b1-81c5a912a579/fabio/alloc/fabio.properties"
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.080Z [INFO]  client.gc: marking allocation for GC: alloc_id=e1ebf603-838a-a1bd-54b1-81c5a912a579
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.216Z [WARN]  client.alloc_runner.task_runner.task_hook: failed to reattach to logmon process: alloc_id=2c506767-a2c3-1b8c-18a3-0e94ab024408 task=connect-proxy-count-api error="Reattachment process not found"
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.216Z [WARN]  client.alloc_runner.task_runner.task_hook: failed to reattach to logmon process: alloc_id=2c506767-a2c3-1b8c-18a3-0e94ab024408 task=web error="Reattachment process not found"
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.219Z [WARN]  client.alloc_runner.task_runner.task_hook: failed to reattach to logmon process: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e task=connect-proxy-count-dashboard error="Reattachment process not found"
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.219Z [WARN]  client.alloc_runner.task_runner.task_hook: failed to reattach to logmon process: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e task=dashboard error="Reattachment process not found"
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.227Z [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=2c506767-a2c3-1b8c-18a3-0e94ab024408 task=connect-proxy-count-api path=/opt/nomad/alloc/2c506767-a2c3-1b8c-18a3-0e94ab024408/alloc/logs/.connect-proxy-count-api.stdout.fifo @module=logmon timestamp=2020-03-28T18:20:19.226Z
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.228Z [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=2c506767-a2c3-1b8c-18a3-0e94ab024408 task=connect-proxy-count-api @module=logmon path=/opt/nomad/alloc/2c506767-a2c3-1b8c-18a3-0e94ab024408/alloc/logs/.connect-proxy-count-api.stderr.fifo timestamp=2020-03-28T18:20:19.228Z
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.229Z [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e task=dashboard @module=logmon path=/opt/nomad/alloc/6d4b864b-4ad2-6368-e74f-3bba4161148e/alloc/logs/.dashboard.stdout.fifo timestamp=2020-03-28T18:20:19.229Z
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.229Z [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e task=dashboard @module=logmon path=/opt/nomad/alloc/6d4b864b-4ad2-6368-e74f-3bba4161148e/alloc/logs/.dashboard.stderr.fifo timestamp=2020-03-28T18:20:19.229Z
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.230Z [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=2c506767-a2c3-1b8c-18a3-0e94ab024408 task=web @module=logmon path=/opt/nomad/alloc/2c506767-a2c3-1b8c-18a3-0e94ab024408/alloc/logs/.web.stdout.fifo timestamp=2020-03-28T18:20:19.230Z
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.230Z [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=2c506767-a2c3-1b8c-18a3-0e94ab024408 task=web @module=logmon path=/opt/nomad/alloc/2c506767-a2c3-1b8c-18a3-0e94ab024408/alloc/logs/.web.stderr.fifo timestamp=2020-03-28T18:20:19.230Z
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.249Z [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e task=connect-proxy-count-dashboard path=/opt/nomad/alloc/6d4b864b-4ad2-6368-e74f-3bba4161148e/alloc/logs/.connect-proxy-count-dashboard.stdout.fifo @module=logmon timestamp=2020-03-28T18:20:19.249Z
Mar 28 18:20:19 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:19.249Z [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e task=connect-proxy-count-dashboard @module=logmon path=/opt/nomad/alloc/6d4b864b-4ad2-6368-e74f-3bba4161148e/alloc/logs/.connect-proxy-count-dashboard.stderr.fifo timestamp=2020-03-28T18:20:19.249Z
...

Mar 28 18:20:23 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:23.081Z [WARN]  client.alloc_runner.task_runner.task_hook.logmon.nomad: timed out waiting for read-side of process output pipe to close: alloc_id=e1ebf603-838a-a1bd-54b1-81c5a912a579 task=fabio @module=logmon timestamp=2020-03-28T18:20:23.081Z
Mar 28 18:20:23 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:23.081Z [WARN]  client.alloc_runner.task_runner.task_hook.logmon.nomad: timed out waiting for read-side of process output pipe to close: alloc_id=e1ebf603-838a-a1bd-54b1-81c5a912a579 task=fabio @module=logmon timestamp=2020-03-28T18:20:23.081Z
Mar 28 18:20:23 sb-sand-nomadagent1 nomad[18885]:     2020/03/28 18:20:23.084087 [INFO] (runner) stopping
Mar 28 18:20:23 sb-sand-nomadagent1 nomad[18885]:     2020/03/28 18:20:23.084413 [INFO] (runner) received finish
Mar 28 18:20:25 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:20:25.454Z [INFO]  client: node registration complete
Mar 28 18:20:27 sb-sand-nomadagent1 nomad[18885]: ==> Newer Nomad version available: 0.10.5 (currently running: 0.10.4)
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.761Z [ERROR] client.alloc_runner.task_runner.task_hook.stats_hook: failed to start stats collection for task with unrecoverable error: alloc_id=2c506767-a2c3-1b8c-18a3-0e94ab024408 task=web error="container stopped"
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.764Z [ERROR] client.alloc_runner.task_runner.task_hook.stats_hook: failed to start stats collection for task with unrecoverable error: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e task=dashboard error="container stopped"
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.765Z [ERROR] client.alloc_runner.task_runner.task_hook.stats_hook: failed to start stats collection for task with unrecoverable error: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e task=connect-proxy-count-dashboard error="container stopped"
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.766Z [ERROR] client.alloc_runner.task_runner.task_hook.stats_hook: failed to start stats collection for task with unrecoverable error: alloc_id=2c506767-a2c3-1b8c-18a3-0e94ab024408 task=connect-proxy-count-api error="container stopped"
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.780Z [WARN]  client.driver_mgr.docker: RemoveImage on non-referenced counted image id: driver=docker image_id=sha256:920eaa8f01535d374004289a5f728a890114b3bd38ae71859c50a2c447daf1db
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.783Z [WARN]  client.driver_mgr.docker: RemoveImage on non-referenced counted image id: driver=docker image_id=sha256:72e91d8680d853b874d9aedda3a4b61048630d2043dd490ff36f5b0929f69874
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.786Z [WARN]  client.driver_mgr.docker: RemoveImage on non-referenced counted image id: driver=docker image_id=sha256:729a950cfecc3f00d00013e90d9d582a6c72fc5806376b1a342c34270ffd8113
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.786Z [WARN]  client.driver_mgr.docker: RemoveImage on non-referenced counted image id: driver=docker image_id=sha256:72e91d8680d853b874d9aedda3a4b61048630d2043dd490ff36f5b0929f69874
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.793Z [INFO]  client.gc: marking allocation for GC: alloc_id=2c506767-a2c3-1b8c-18a3-0e94ab024408
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.795Z [ERROR] client.alloc_runner.runner_hook: failed to cleanup network for allocation, resources may have leaked: alloc_id=2c506767-a2c3-1b8c-18a3-0e94ab024408 alloc=2c506767-a2c3-1b8c-18a3-0e94ab024408 error="cni plugin not initialized"
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.796Z [INFO]  client.gc: marking allocation for GC: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e
Mar 28 18:21:55 sb-sand-nomadagent1 nomad[18885]:     2020-03-28T18:21:55.799Z [ERROR] client.alloc_runner.runner_hook: failed to cleanup network for allocation, resources may have leaked: alloc_id=6d4b864b-4ad2-6368-e74f-3bba4161148e alloc=6d4b864b-4ad2-6368-e74f-3bba4161148e error="cni plugin not initialized"


```

**Comment by spuder** [comment]
Easy way to reproduce this. Add a token to the Nomad-Agent and start a job that uses consul-connect

`/etc/nomad/config.json`
```
{
  "consul": {
    'token": "123456"
  }
...
```

Then, without stopping the job; remove the token and restart the nomad agent
```
vim /etc/nomad/config.json
service nomad restart
```

Then stop the job, you will see in the logs

```
Mar 29 01:51:21 sb-sand-nomadagent1 nomad[12431]:     2020-03-29T01:51:21.336Z [ERROR] client.alloc_runner.task_runner.task_hook.stats_hook: failed to start stats collection for task with unrecoverable error: alloc_id=c689661c-41f6-4589-d3f6-5e71aded9e7c task=connect-proxy-count-dashboard error="container stopped"
Mar 29 01:51:21 sb-sand-nomadagent1 nomad[12431]:     2020-03-29T01:51:21.337Z [WARN]  client.driver_mgr.docker: RemoveImage on non-referenced counted image id: driver=docker image_id=sha256:72e91d8680d853b874d9aedda3a4b61048630d2043dd490ff36f5b0929f69874
Mar 29 01:51:21 sb-sand-nomadagent1 nomad[12431]:     2020-03-29T01:51:21.338Z [WARN]  client.driver_mgr.docker: RemoveImage on non-referenced counted image id: driver=docker image_id=sha256:920eaa8f01535d374004289a5f728a890114b3bd38ae71859c50a2c447daf1db
Mar 29 01:51:21 sb-sand-nomadagent1 nomad[12431]:     2020-03-29T01:51:21.338Z [WARN]  client.driver_mgr.docker: RemoveImage on non-referenced counted image id: driver=docker image_id=sha256:729a950cfecc3f00d00013e90d9d582a6c72fc5806376b1a342c34270ffd8113
Mar 29 01:51:21 sb-sand-nomadagent1 nomad[12431]:     2020-03-29T01:51:21.346Z [INFO]  client.gc: marking allocation for GC: alloc_id=9e2e3563-fe7a-0938-6b73-61655381c532
Mar 29 01:51:21 sb-sand-nomadagent1 nomad[12431]:     2020-03-29T01:51:21.348Z [ERROR] client.alloc_runner.runner_hook: failed to cleanup network for allocation, resources may have leaked: alloc_id=9e2e3563-fe7a-0938-6b73-61655381c532 alloc=9e2e3563-fe7a-0938-6b73-61655381c532 error="cni plugin not initialized"
Mar 29 01:51:21 sb-sand-nomadagent1 nomad[12431]:     2020-03-29T01:51:21.354Z [WARN]  client.driver_mgr.docker: RemoveImage on non-referenced counted image id: driver=docker image_id=sha256:72e91d8680d853b874d9aedda3a4b61048630d2043dd490ff36f5b0929f69874
Mar 29 01:51:21 sb-sand-nomadagent1 nomad[12431]:     2020-03-29T01:51:21.357Z [INFO]  client.gc: marking allocation for GC: alloc_id=c689661c-41f6-4589-d3f6-5e71aded9e7c
Mar 29 01:51:21 sb-sand-nomadagent1 nomad[12431]:     2020-03-29T01:51:21.359Z [ERROR] client.alloc_runner.runner_hook: failed to cleanup network for allocation, resources may have leaked: alloc_id=c689661c-41f6-4589-d3f6-5e71aded9e7c alloc=c689661c-41f6-4589-d3f6-5e71aded9e7c error="cni plugin not initialized"
```

Obviously you shouldn't be removing tokens on the fly, but sometimes you need to rotate a token.

**Comment by shoenig** [comment]
Should be fixed as part of #7537
